src/porepy/utils/grid_refinement.py

```diff
@@ -0,0 +1,239 @@
+
+    # Create sps.csc_matrix mapping coarse cells to fine cell centers
+    indptr = np.array([0])
+    indices = np.empty(0)
+
+    cells_ref = g_ref.cell_centers.copy()  # Cell centers in fine grid
+    test_cells_ptr = np.arange(g_ref.num_cells)  # Pointer to cell centers
+    nodes = g.nodes.copy()
+
+    if g.dim == 1:
+        nodes = nodes.copy()
+        tangent = pp.map_geometry.compute_tangent(nodes)
+        reference = [1, 0, 0]
+        R = pp.map_geometry.project_line_matrix(nodes, tangent, tol=tol, reference=reference)
+        nodes = R.dot(nodes)[0, :]
+        cells_ref = R.dot(cells_ref)[0, :]
+
+    elif g.dim == 2:  # Pre-processing for efficiency
+        nodes = nodes.copy()
+        R = pp.map_geometry.project_plane_matrix(nodes, check_planar=False)
+        nodes = np.dot(R, nodes)[:2, :]
+        cells_ref = np.dot(R, cells_ref)[:2, :]
+
+    # Loop through every coarse cell
+    for st, nd in slices:
+
+        nodes_idx = cell_nodes.indices[st:nd]
+        num_nodes = nodes_idx.size
+
+        if g.dim == 1:
+            assert (num_nodes == 2)
+            line = np.sort(nodes[nodes_idx])
+            test_points = cells_ref[test_cells_ptr]
+            in_poly = np.searchsorted(line, test_points, side='left') == 1
+
+        elif g.dim == 2:
+            assert (num_nodes == 3), "We assume simplexes in 2D (i.e. 3 nodes)"
+            polygon = nodes[:, nodes_idx]
+            test_points = cells_ref[:, test_cells_ptr]
+            in_poly = pp.geometry_property_checks.point_in_polygon(
+                polygon, test_points, tol=tol)
+
+        elif g.dim == 3:
+            # Make polyhedron from node coordinates
+            # Polyhedron defined as a list of nodes defining its (convex) faces.
+            # Assumes simplexes: Every node except one defines every face.
+            assert (num_nodes == 4), "We assume simplexes in 3D (i.e. 4 nodes)"
+            node_coords = nodes[:, nodes_idx]
+
+            ids = np.arange(num_nodes)
+            polyhedron = [node_coords[:, ids != i] for i in np.arange(num_nodes)]
+            test_points = cells_ref[:, test_cells_ptr]  # Test only points not inside another polyhedron.
+            in_poly = pp.geometry_property_checks.point_in_polyhedron(
+                polyhedron=polyhedron, test_points=test_points, tol=tol
+            )
+
+        else:
+            logger.warning(f"A grid of dimension {g.dim} encountered. Skip!")
+            continue
+
+        # Update pointer to which cell centers to use as test points
+        in_poly_ids = test_cells_ptr[in_poly]  # id of cells inside this polyhedron
+        test_cells_ptr = test_cells_ptr[~in_poly]  # Keep only cells not inside this polyhedron
+
+        # Update mapping
+        indices = np.append(indices, in_poly_ids)
+        indptr = np.append(indptr, indptr[-1] + in_poly_ids.size)
+
+    data = np.ones(indices.size)
```

Comment:
do you want to have a measure ration between the reference and the mapped one? just proposing, maybe it's out of scope

Reply:
@alessiofumagalli I have no idea what you mean here.

Reply:
it took me a while to understand it for me as well.. I mean, a projection matrix that map variables from the original grid to the refined one (maybe the vice versa, as well)